Diagnostic/diagnostic.py

```diff
@@ -879,13 +905,19 @@ def install_omi():
             hutil.error("OMI install failed " + str(maxTries) + " times. Giving up...")
             return 1, "OMI install failed " + str(maxTries) + " times"
 
+    shouldRestartOmi = False
+
+    # Check if OMI is configured to listen to 1270 and reconfigure if so.
+    omiListensTo1270 = RunGetOutput("grep '^httpsport=0,1270' /etc/opt/omi/conf/omiserver.conf")[0] is 0
```

Comment:
there are several  issues here: if OMI change port 1270 to 1271, this will fail; .if OMI add extra space like 0, 1270. this will fail too.

One way to do is: 1) search httpsport line. 2) check whether it has only 0 or not, if not, make it to be port 0 only.